### DISCUSSION TITLE
Possible summary of changes format

### DISCUSSION THREAD

**Discussion by vvbandeira** [The-OpenROAD-Project/OpenROAD#3939] [General]
While discussion #3856 has leaned to whether there should be a release and how frequent it should be, this discussion thread is to help guide the format and the process we would use to do such releases. We are considering using an automated way to generate "release notes" or at least a summary of recent changes. One proposed way to achieve this is to use PR titles and "Merge commit" messages. We would define and enforce a style of PR title, e.g., `[<TYPE>] <TOOL>: <SUMMARY>` where `[FEAT] drt: added shiny router cmd`. While `<TOOL>` is well defined (i.e., any module under `src/third-party` and `<SUMMARY>` will depend on the PR, what would be the most relevant values `<TYPE>` to include and their definition? As a starter point:

- `[FEAT]` or `[NEW]`: for when we add new features, either in the form of Tcl/Python API changes -- e.g., new commands or flags.
- `[DOCS]` or `[DOC]`: for documentation only changes.
- `[BUGF]` or `[FIX]`: when fixing bugs or issues.
- `[UPGR]` or `[ENH]`: upgrade/enhance -- not very sure about this, but it was what I thought with 4/3 letters to keep similar to above, mostly a self-imposed restriction, open to feedback.

cc/ @maliberty @dralabeing @luarss @rovinski @QuantamHD

**Comment by maliberty** [comment]
I don't think of it as "summary of recent changes" as we still don't plan to do releases.  <TOOL> is also optional as some changes may span multiple tools though that should be less common.

**Comment by rovinski** [comment]
I like the spirit, but I can foresee a few issues:

1. A PR might span multiple tools, in which case it's hard to accommodate the format
2. A PR might implement a feature only partially, in which case it's not ready for full use yet
3. A PR might span multiple types (e.g. both feature and enhancement).
4. If we want this scheme to stick, we will likely need a bot to enforce it or at least remind PR authors

As such,

* I think that enhancement and feature should probably just be one tag. I think PRs end up being both too often to distinguish
* We need some sort of mechanism or identifying a feature as not ready for general use yet.
